subworkflows/local/03_assemble/assemble_organelles.nf

```diff
@@ -1,40 +1,74 @@
 
 workflow ASSEMBLE_ORGANELLES {
     take:
-    ch_raw_assemblies
+    ch_input
+    ch_input_mode
 
     main:
     ch_versions = Channel.empty()
     // Find mitochondria
     // TODO: Need to check options to mitohifi modules.
-    MITOHIFI_FINDMITOREFERENCE( ch_raw_assemblies.map { meta, _assemblies -> [ meta, meta.sample.name ] }.unique() )
-    mitohifi_ch = ch_raw_assemblies
-        .combine(
-            MITOHIFI_FINDMITOREFERENCE.out.fasta
-                .join(MITOHIFI_FINDMITOREFERENCE.out.gb),
-            by: 0
-        )
-        .multiMap { meta, assembly, mitofa, mitogb ->
-            input: [ meta, assembly.pri_fasta ]
-            reference: mitofa
-            genbank: mitogb
-            mito_code: meta.sample.mito_code
+    MITOHIFI_FINDMITOREFERENCE( ch_input.map { meta, _input -> [ meta, meta.sample.name ] }.unique() )
+
+    // Validate input mode and run MITOHIFI accordingly on contigs or reads
+    if (["c", "r"].contains(ch_input_mode)) {
+        if (ch_input_mode == "c") {
+            mitohifi_ch = ch_input
+                .combine(
+                    MITOHIFI_FINDMITOREFERENCE.out.fasta
+                        .join(MITOHIFI_FINDMITOREFERENCE.out.gb),
+                    by: 0
+                )
+                .multiMap { meta, assembly, mitofa, mitogb ->
+                    input: [ meta, assembly.pri_fasta ]
+                    reference: mitofa
+                    genbank: mitogb
+                    mito_code: meta.sample.mito_code
+                }
+            MITOHIFI_MITOHIFI_CONTIGS(
+                mitohifi_ch.input,
+                mitohifi_ch.reference,
+                mitohifi_ch.genbank,
+                ch_input_mode,
+                mitohifi_ch.mito_code,
+            )
+            ch_versions = ch_versions.mix(
+                MITOHIFI_FINDMITOREFERENCE.out.versions.first(),
+                MITOHIFI_MITOHIFI_CONTIGS.out.versions.first()
+            )
+        } else if (ch_input_mode == "r") {
+            mitohifi_ch = ch_input
+                .combine(
+                    MITOHIFI_FINDMITOREFERENCE.out.fasta
+                        .join(MITOHIFI_FINDMITOREFERENCE.out.gb),
+                    by: 0
+                )
+                .multiMap { meta, reads, mitofa, mitogb ->
+                    input: [ meta, reads ]
+                    reference: mitofa
+                    genbank: mitogb
+                    mito_code: meta.sample.mito_code
+                }
+            MITOHIFI_MITOHIFI_READS(
+                mitohifi_ch.input,
+                mitohifi_ch.reference,
+                mitohifi_ch.genbank,
+                ch_input_mode,
+                mitohifi_ch.mito_code,
+            )
+            ch_versions = ch_versions.mix(
+                MITOHIFI_FINDMITOREFERENCE.out.versions.first(),
+                MITOHIFI_MITOHIFI_READS.out.versions.first()
+            )
```

Comment:
This is basically duplicated code. You could move some of this to the workflow call to remove the need for the `if`.

Reply:
In the `modules.config`, you test the mode (whatever the parameter under `input:` is called) and supply flags.
```
    ext.args = { [
        "--always-on",
        mode in ['c','r'] ? "--opt1" : "",
        mode in ['c'] ? "--opt2" : "",
        mode in ['r'] ? "--opt3" : "--opt4",
    ].minus("").join(" ") }
```
and the user simply overrides the input supplying the opts they want.